pkg/backup/compressor.go

```diff
@@ -0,0 +1,34 @@
+package backup
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+)
+
+// Compress compresses a file using gzip.
+func Compress(filepath string) (string, error) {
+	data, err := os.ReadFile(filepath)
+	if err != nil {
+		return "", fmt.Errorf("read all: %s", err)
+	}
+	var b bytes.Buffer
+	w, err := gzip.NewWriterLevel(&b, gzip.BestCompression)
+	if err != nil {
+		return "", fmt.Errorf("new writer level: %s", err)
+	}
+	if _, err := w.Write(data); err != nil {
+		return "", fmt.Errorf("gzip write: %s", err)
+	}
+	if err := w.Close(); err != nil {
+		return "", fmt.Errorf("close: %s", err)
+	}
+
+	newFilepath := fmt.Sprintf("%s.gz", filepath)
+	if err := os.WriteFile(newFilepath, b.Bytes(), 0o755); err != nil {
+		return "", fmt.Errorf("write file: %s", err)
+	}
```

Comment:
Makes sense. I thought this would be a problem, but I wasn't sure if the `gzip` compression would work in stream mode (if it has to read the whole file beforehand or not). Since I did not find with a simple Google search, I went with the whole file in memory approach.